iceoryx_posh/source/mepoo/chunk_settings.cpp

```diff
@@ -66,18 +66,10 @@ expected<ChunkSettings, ChunkSettings::Error> ChunkSettings::create(const uint32
     uint64_t requiredChunkSize =
         calculateRequiredChunkSize(userPayloadSize, adjustedUserPayloadAlignment, userHeaderSize);
 
-    if (requiredChunkSize > std::numeric_limits<uint32_t>::max())
-    {
-        return err(ChunkSettings::Error::REQUIRED_CHUNK_SIZE_EXCEEDS_MAX_CHUNK_SIZE);
-    }
-
-    return ok(ChunkSettings{userPayloadSize,
-                            adjustedUserPayloadAlignment,
-                            userHeaderSize,
-                            adjustedUserHeaderAlignment,
-                            static_cast<uint32_t>(requiredChunkSize)});
+    return ok(ChunkSettings{
+        userPayloadSize, adjustedUserPayloadAlignment, userHeaderSize, adjustedUserHeaderAlignment, requiredChunkSize});
 }
-uint64_t ChunkSettings::calculateRequiredChunkSize(const uint32_t userPayloadSize,
+uint64_t ChunkSettings::calculateRequiredChunkSize(const uint64_t userPayloadSize,
```

Comment:
Here we definitely needs  changes since the calculation could easily result in an overflow. It was easier when the input parameter were `uint32_t`.

Luckily we already moved to the static creation method pattern instead of calling the ctor explicitly and therefore we can also use an `iox::expected` here and forward the error in the `create` method

Reply:
@elBoberido I think having an overflow here is most likely caused by just a programming mistake, and therefore I would vote to use `IOX_ENSURES` here and panic when this occurs. I cannot imagine a use case where an overflow can happen, and it is a real world example.

Reply:
I would avoid to panic here, an incorrect size is not different from an incorrect alignment, which also does not panic. Let's keep doing it like the local error handling on the class instead of mixing it. It makes testing also a little bit easier.